### DISCUSSION TITLE
How to implement a health check that will check all tenants together?

### DISCUSSION THREAD

**Discussion by xperiandri** [OrchardCMS/OrchardCore#14462] [Q&A]
My idea is to run health checks per each tenant using `shellScope.UsingAsync` and combine the results. Is it good?

**Comment by hishamco** [comment]
I already did something similar

https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules/blob/main/src/OrchardCoreContrib.HealthChecks/README.md

https://www.youtube.com/watch?v=GeV_n7e4Niw
